### DISCUSSION TITLE
random prompt doesn't change inside of a batch

### DISCUSSION THREAD

**Discussion by bobpuffer1** [adieyal/comfyui-dynamicprompts#7] [Q&A]
Set up a batch of 8 And indicate a wild card of hair color
there are eight hair colors in the wild card text file
random prompts uses the same hair color for all 8 images in a batch
This means in order to randomize a batch size needs to be 1

**Comment by bobpuffer1** [comment]
NM
It doesn't appear it's using the wildcards file at all.

**Comment by bobpuffer1** [comment]
Knowing that Dynamic Prompts  accesses individual members of a wildcard  file alphabetically it selects the first member (auburn) but never goes  further. A batch of four gives four auburn-haired. Eight batches of one gives eight auburn-haired. The PNG on the README doesn't load  so I don't have an example to observe -- I'm flying blind. Do you know of a tutorial on comfyui dynamic prompts?

**Comment by bobpuffer1** [comment]
I found your samples and tried them and I think it is working. I'll have to play some more. Thannks.

**Comment by adieyal** [comment]
What were you doing before that was different? It's clear that the documentation should be improved (or perhaps there should be a way of preventing the user from doing the wrong thing).

**Comment by bobpuffer1** [comment]
i'm not sure why they seem to be working now. it could be just a better understanding on my part. the prompt shouldn't be revisited during a batch. it should be revisited and randomized between each batch Unless it's combinatorial. as far as I'm concerned combinatorial is broken because of alphabetizes before it randomizes so it's pseudo random and restart between batches or whatever generational parameters or input the fact that it alphabetizes is okay but it shouldn't restart.
